plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -465,17 +465,19 @@ createWidget("discourse-poll-pie-canvas", {
     loadScript("/javascripts/Chart.min.js").then(() => {
       const data = attrs.poll.options.mapBy("votes");
       const labels = attrs.poll.options.mapBy("html");
-      const config = pieChartConfig(data, labels);
+      const config = pieChartConfig(data, labels, {
+        legendContainerId: `poll-results-legend-${attrs.id}`,
+      });
 
       const el = document.getElementById(`poll-results-chart-${attrs.id}`);
-      // eslint-disable-next-line
-      let chart = new Chart(el.getContext("2d"), config);
-      document.getElementById(
-        `poll-results-legend-${attrs.id}`
-      ).innerHTML = chart.generateLegend();
+      this._chart = new Chart(el.getContext("2d"), config);
     });
   },
 
+  willRerenderWidget() {
+    this._chart?.destroy();
```

Comment:
we were not reseting the canvas which should be done, and is fatal error in 3.x anyways